app/services/album.js

```diff
@@ -25,3 +26,30 @@ exports.photos = albumId =>
     .catch(() => {
       throw errors.externalApiError('Error in external API');
     });
+
+exports.buy = async (albumId, currentUser) => {
+  try {
+    const response = await axios.get(`${apiUrl}/albums/${albumId}`);
+    let album = await models.album.findOne({ where: { title: response.data.title } });
+    if (!album) {
+      album = await models.album.create({
+        title: response.data.title,
+        userId: currentUser.dataValues.id
+      });
+    }
+    const boughtAlbum = await models.userAlbums.findOne({
+      where: { albumId: album.id, userId: currentUser.id }
+    });
```

Comment:
Same with these. If they fail the catch will simply forward the error as is, without wrapping it in our errors object.

Reply:
Also, I wont ask you to change it, but there was no need to have an albums model. The `albums_user` model was enough, since the`album_id`s in that table could have simply referenced the ones of the service.
As it is it doesn't really add value, It's not working as a cache because you always call the external service first.